go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/containers/podman/v4
 
-go 1.17
+go 1.18
```

Comment:
I could be wrong but I always thought this is the minimal version the code compiles on. Only when we add go 1.18+ features directly or in dependencies this must be changed.
cc @vrothberg

Reply:
As far as I know this is only for upstream dependencies. If there are such dependencies, then this change obviously needs more discussion.

For more context, we were looking to update the golang version for the `container-device-interface` which is used as a dependency here and I thought I would bump the version before we did so so as to not cause other issues when a new version is pulled in.

Reply:
Thanks for being so pro-active, @elezar!  Since Go 1.17 is EOL, we are good to bump the version but as @Luap99 mentioned, we usually do it only when needed; and the need is just ahead :)

But the change seems to impact the linting, so you may need to wrestle through the `make validate` errors, @elezar.  If you want us to do that, please let us know and we can kick the PR over the finish line.

Reply:
I will have a look at the linting errors next week and see what's required

Reply:
I'm having some trouble with my local `make validate` results. When I run these I see a number of `typecheck` failures due to unused imports whereas these are not reported in the CI. Is there somthing to note when running these locally?

Reply:
I don't think so, if you use the same golangci-lint version as used in CI it should work locally.

Reply:
I can reproduce locally, weird.

Reply:
Running the `make validate-in-container` target from #16613 with these changes applied shows a number of typedef errors that I didn't see in CI (Unless I'm misreading the logs)